### PR TITLE
TMS-914: Add new anchor-links block & functionalities

### PR DESCRIPTION
**Pull request by eebbi**
Severa-ID: 2108
Severa-kuvaus: TMS-914 8 - Ankkurilinkit
Task: https://hiondigital.atlassian.net/browse/TMS-914

## Description
- Add new anchor-links block & functionalities
- Add anchor id-elements to blocks & components

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: assets/scripts/anchor-links.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ *  Copyright (c) 2023. Hion Digital
+ */
```
Comment (vloy) [inline assets/scripts/anchor-links.js:3]: Turhaa tauhkaa?

File: assets/styles/blocks/custom/_anchor-links.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.anchor-links {
+    &__list {
+        padding-left: 1rem;
```
Comment (vloy) [inline assets/styles/blocks/custom/_anchor-links.scss:3]:
```suggestion
.anchor-links {
    &__list {
        padding-left: $theme-spacing;
```
?

File: partials/blocks/block-image-banner.dust [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {?image}
-    <div class="image-banner mt-4 mb-4 mt-8-tablet mb-8-tablet is-relative is-align-full has-background-cover"
-         style="background-image: url({image.sizes.large})">
+    <div {?anchor}id="{anchor|attr}"{/anchor}
+         class="image-banner mt-4 mb-4 mt-8-tablet mb-8-tablet is-relative is-align-full has-background-cover"
+         style="background-image: url({image.sizes.large|attr})">
```
Comment (vloy) [inline partials/blocks/block-image-banner.dust:4]:
```suggestion
         style="background-image: url({image.sizes.large|url})">
```

Comment (vloy) [reply partials/blocks/block-image-banner.dust:4]: Eikös tää palauta urlin?

File: partials/blocks/block-image-gallery.dust [hunk 4]
```diff
@@ -1,4 +1,6 @@
-<div class="image-gallery {align_class} gallery {column_class} js-image-gallery mt-8 mb-8 mt-11-desktop mb-11-desktop"
+<div
+     {?anchor}id="{anchor|attr}"{/anchor}
+     class="image-gallery {align_class} gallery {column_class} js-image-gallery mt-8 mb-8 mt-11-desktop mb-11-desktop"
```
Comment (vloy) [inline partials/blocks/block-image-gallery.dust:3]:
```suggestion
     class="image-gallery {align_class|attr} gallery {column_class|attr} js-image-gallery mt-8 mb-8 mt-11-desktop mb-11-desktop"
```

File: partials/blocks/block-image-banner.dust (hunk 3 above)
Comment (eebbi) [reply partials/blocks/block-image-banner.dust:4]: Palauttaahan se :) korjataan

File: assets/scripts/anchor-links.js (hunk 1 above)
Comment (eebbi) [reply assets/scripts/anchor-links.js:3]: Eikai turhaa, mutta väärä vuosi lipsahtanut

**Review by vloy — APPROVED** [review]
🙆‍♂️

**Pull request merged**